src/test/Profiling.cpp

```diff
@@ -110,14 +103,19 @@ class Profiling : public Test {
 
 	// Print timings of all functions
 	void log_timings() const {
-		std::stringstream ss;
-        ss << "\nFunction timings:\n";
-        for (const auto& [name, duration] : timings) {
-            ss << name << " took " << duration.count() / 1000.0 / average << " ms (" << duration.count() / average << " µs).\n";
-        }
-        ss << "Total time: " << this->total_time.count() / 1000.0 / average << " ms (" << this->total_time.count() / average << " µs)\n";
-        ss << "Amount of gameobjects: " << game_object_count << "\n";
-        GTEST_LOG_(INFO) << ss.str();
+		std::string result = "\nFunction timings:\n";
+    
+    for (const auto& [name, duration] : timings) {
+        result += name + " took " + std::to_string(duration.count() / 1000.0 / average) + " ms (" +
+                  std::to_string(duration.count() / average) + " µs).\n";
+    }
+
+    result += "Total time: " + std::to_string(this->total_time.count() / 1000.0 / average) + " ms (" +
+              std::to_string(this->total_time.count() / average) + " µs)\n";
+    
+    result += "Amount of gameobjects: " + std::to_string(game_object_count) + "\n";
+
+    GTEST_LOG_(INFO) << result;
```

Comment:
Jouw code bevat unicode en dat mag niet volgens de code standaard. Deze code kan trouwens veel netter met `std::format` omdat die ingebouwde functies heeft om `std::chrono::duration` om te zetten naar een string met een grootheid.
```cpp
	string fmt_duration(const chrono::microseconds & duration) const {
		using chrono::duration_cast;
		using ms = chrono::milliseconds;
		using us = chrono::microseconds;
		return format("{} ({})", duration_cast<us>(duration), duration_cast<ms>(duration));
	}

	// Print timings of all functions
	void log_timings() const {
		string result = "Function timings:\n";

		for (const auto & [name, duration] : timings) {
			result += format("{} took {}\n", name, fmt_duration(duration));
		}

		result += format("Total time: {}\n", fmt_duration(total_time));
		result += format("Amount of gameobjects: {}\n", game_object_count);

		GTEST_LOG_(INFO) << result;
	}
```

Reply:
Zoals besproken is ook hiervoor een ticket aangemaakt in miro voor als er tijd over is na de functionele engine features